vaadin-grid-flow-demo/src/main/java/com/vaadin/flow/component/grid/demo/GridDemo.java

```diff
@@ -2059,6 +2073,322 @@ private void createNotBufferedDynamicEditor() {
+        grid2.setDropMode(DropMode.BETWEEN);
+        grid2.setColumns("firstName", "lastName");
+
+        // end-source-example
+
+        HorizontalLayout hl = new HorizontalLayout(grid, grid2);
+
+        addCard("Drag and Drop", "Drag Rows Between Grids", hl);
+    }
+
+    private void createDropLocations() {
+
+        // begin-source-example
+        // source-example-heading: Drop Location
+        PersonService personService = new PersonService();
+
+        Grid<Person> grid = new Grid<>(Person.class);
+        grid.setItems(personService.fetch(0, 50));
+
+        grid.addDragStartListener(
+                event -> draggedItems = event.getDraggedItems());
+        grid.setColumns("firstName", "lastName", "phoneNumber");
+        grid.setRowsDraggable(true);
+
+        TreeData<Person> td = new TreeData<>();
+        td.addItems(null, personService.fetch(51, 2));
+
+        TreeGrid<Person> treeGrid = new TreeGrid<>();
+        treeGrid.setDataProvider(new TreeDataProvider<Person>(td));
+        /*
+         * This enables dropping in between or on top of the existing grid rows.
+         */
+        treeGrid.setDropMode(DropMode.ON_TOP_OR_BETWEEN);
+        treeGrid.addHierarchyColumn(Person::getfirstName)
+                .setHeader("firstName");
+        treeGrid.addColumn(Person::getLastName).setHeader("lastName");
+        treeGrid.addColumn(Person::getPhoneNumber).setHeader("phoneNumber");
+        treeGrid.addDropListener(event -> {
+            // Remove the items from the source grid
+            ListDataProvider<Person> sourceDataProvider = (ListDataProvider<Person>) grid
+                    .getDataProvider();
+            List<Person> sourceItems = new ArrayList<Person>(
+                    sourceDataProvider.getItems());
+            sourceItems.remove(draggedItems.get(0));
+            grid.setItems(sourceItems);
+
+            // Add the item to target grid
+            Optional<Person> dropOverItem = event.getDropTargetRow();
+            if (DropLocation.ON_TOP.equals(event.getDropLocation())) {
+                td.addItem(dropOverItem.get(), draggedItems.get(0));
+            } else {
+                Person parent = td.getParent(dropOverItem.get());
+                td.addItem(parent, draggedItems.get(0));
+
+                List<Person> siblings = td.getChildren(parent);
+                int dropIndex = td.getChildren(parent)
+                        .indexOf(dropOverItem.get()) - 1
+                        + (event.getDropLocation() == DropLocation.BELOW ? 1
+                                : 0);
+                td.moveAfterSibling(draggedItems.get(0),
+                        siblings.get(dropIndex));
+            }
+
+            treeGrid.getDataProvider().refreshAll();
+
+        });
+
+        // end-source-example
+
+        HorizontalLayout hl = new HorizontalLayout(grid, treeGrid);
+
+        addCard("Drag and Drop", "Drop Location", hl);
+    }
+
+    private void createDragData() {
```

Comment:
There could be a comment somewhere to explain that this allows dragging between applications

Reply:
Done